src/sketcher/sketchermodel.ts

```diff
@@ -0,0 +1,233 @@
+import { showErrorMessage } from '@jupyterlab/apputils';
+import { v4 as uuid } from 'uuid';
+
+import { IDict, IJupyterCadDoc } from '../types';
+import { IJCadObject } from '../_interface/jcad';
+import { IGeomCircle, IGeomLineSegment } from '../_interface/sketch';
+
+import { distance } from './helper';
+import { Line } from './elements/line';
+import { Point } from './elements/point';
+import {
+  ICircle,
+  ILine,
+  IOperator,
+  IPlane,
+  IPoint,
+  IPosition,
+  ISketcherModel
+} from './types';
+import { Circle } from './elements/circle';
+
+export class SketcherModel implements ISketcherModel {
+  constructor(options: { gridSize: number; sharedModel?: IJupyterCadDoc }) {
+    this._gridSize = options.gridSize;
+    this._sharedModel = options.sharedModel;
+  }
+
+  get gridSize(): number {
+    return this._gridSize;
+  }
+  get points(): Map<string, IPoint> {
+    return this._points;
+  }
+  get lines(): Map<string, ILine> {
+    return this._lines;
+  }
+  get circles(): Map<string, ICircle> {
+    return this._circles;
+  }
+  get editing(): { type: IOperator | null; content: IDict | null } {
+    return this._editing;
+  }
+  startEdit(type: IOperator, content: IDict): void {
+    this._editing.type = type;
+    this._editing.content = content;
+  }
+  updateEditiing(type: IOperator, content: IDict): void {
```

Comment:
```suggestion
  updateEditing(type: IOperator, content: IDict): void {
```

Reply:
Maybe even `updateEdit`? So that it's more homogeneous with `startEdit` and `stopEdit`